File: 11725.py
n = int(input())
visited = [0] * (n+1)
m = [[] for _ in range(n+1)]
for i in range(n-1):
    a, b = map(int, input().split())
    a-=1
    b-=1
    m[a].append(b)
    m[b].append(a)

q = [0]
visited[0] = 1
ans = [0] *(n+1)


while q:
    s = q[0]
    q.pop(0)
    for p in m[s]:
        if visited[p] == 0:
            visited[p] = 1
            ans[p] = s
            q.append(p)

for i in range(1,n):
    print(ans[i]+1)

File: 11725.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

#define MAX 100001

queue<int> q;
vector<int> map[MAX];
int visited[MAX];

int main(){
  int n;
  cin >> n;

  for (int i = 0; i < n-1; i++)
  {
    int a,b;
    cin >> a >> b;
    map[a].push_back(b);
    map[b].push_back(a);
  }

  vector<int> v(n+1,0);
  visited[1] = 1;
  q.push(1);
  while(!q.empty()){
    int s = q.front();
    q.pop();
    for (auto i = map[s].begin(); i != map[s].end(); i++)
    {
      if(!visited[*i]){
        visited[*i]=1;
        q.push(*i);
        v[*i]=s;
      }
    }
  }
  
  for(int i = 2; i <= n;i++){
    cout << v[i] << '\n';
  }
  return 0;
}